pulser-core/pulser/sequence/sequence.py

```diff
@@ -861,6 +860,15 @@ def build(
         # Deepcopy the base sequence (what remains)
         seq = copy.deepcopy(seq)
 
+        # NOTE: Changes to seq are now safe to do
+
+        if qubits:
+            reg = cast(MappableRegister, self._register).build_register(qubits)
+            self._set_register(seq, reg)
+            assert self.is_register_mappable()
```

Comment:
Why do we need this assertion ? Shouldn't `_set_register` with a `MappableRegister` ensure this ?

Reply:
Yes, I committed this by mistake, sorry...